include/revng/ABI/FunctionType.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Types.h"
+
+namespace abi::FunctionType {
+
+/// Best effort `CABIFunctionType` to `RawFunctionType` conversion.
+///
+/// If `ABI` is not specified, `TheBinaryToAddSupportTypesTo.DefaultABI`
+/// is used instead.
+std::optional<model::CABIFunctionType>
+tryConvertToCABI(const model::RawFunctionType &Function,
+                 model::Binary &TheBinaryToAddSupportTypesTo,
+                 std::optional<model::ABI::Values> ABI = std::nullopt);
+
+/// Best effort `RawFunctionType` to `CABIFunctionType` conversion.
+///
+/// \note: this convertion is loosy since there's no way to represent some types
+/// in `RawFunctionType` in a reversible manner.
+std::optional<model::RawFunctionType>
+tryConvertToRaw(const model::CABIFunctionType &Function,
+                model::Binary &TheBinaryToAddSupportTypesTo);
+
+struct Layout;
+
+/// Extracts the information about argument and return value location layout
+/// from the \param Function.
+Layout layout(const model::RawFunctionType &Function);
+
+/// Extracts the information about argument and return value location layout
+/// from the \param Function.
+Layout layout(const model::CABIFunctionType &Function);
```

Comment:
Promote to `static` "factory" methods. This also saves us from `friend`.

Reply:
Actually, they can just be constructors at this point! They used to return an `std::optional` but since we got rid of that, no real reason for them to be free functions.

Reply:
Done: https://github.com/revng/revng/pull/131/commits/505909f2fca23b0a400bb6c17031fac8481b8472.